pkg/runtime/pubsub/bulksub.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pubsub
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/dapr/dapr/utils"
+
+	contribPubsub "github.com/dapr/components-contrib/pubsub"
+)
+
+const (
+	maxBulkCountKey     string = "maxBulkCount"
+	maxBulkLatencyMsKey string = "maxBulkLatencyMilliSeconds"
+
+	defaultMaxBulkCount     int = 100
+	defaultMaxBulkLatencyMs int = 1 * 1000
+)
+
+// msgWithCallback is a wrapper around a message that includes a callback function
+// that is called when the message is processed.
+type msgWithCallback struct {
+	msg contribPubsub.BulkMessageEntry
+	cb  func(error)
+}
+
+// defaultBulkSubscriber is the default implementation of BulkSubscriber.
+// It is used when the component does not implement BulkSubscriber.
+type defaultBulkSubscriber struct {
+	p contribPubsub.PubSub
```

Comment:
Should we rename it to `pubsub` instead of `p` or else it will result in something like p.p.Subscribe() ?

Reply:
Yeah, this is same as `defaultBulkPublisher` actually, and is pretty common in Go & Dapr code base.
![image](https://user-images.githubusercontent.com/22132944/190328974-d688f2d0-f64f-4dc5-afde-6f0f4080f091.png)

Let me know if you want me to change it, I don't mind much, but I prefer single letter too.
/cc @mukundansundar

Reply:
Haha, yeah I have seen it in go codebase now at various places - but it is very non-intuitive, for me atleast. But, alright I am fine either way going by precedence logic.